### PR TITLE
add opts to repo name

### PR DESCRIPTION
**Pull request by arthurlogilab**

### CODE REVIEW

**Comment by daks** [comment]
Seems good to me :)

File: apt/repositories.sls [hunk 1]
```diff
@@ -30,6 +30,7 @@ debian-archive-keyring:
 {%- set r_arch = '[arch=' ~ args.arch|join(',') ~ ']' if args.arch is defined else '' %}
 {%- set r_url = args.url or default_url %}
 {%- set r_distro = args.distro or 'stable' %}
+{%- set r_opts = args.opts if args.opts is defined else '' %}
```
Comment (javierbertoli) [inline apt/repositories.sls]: As I understand from the [manpage](https://manpages.debian.org/jessie/apt/sources.list.5.en.html), the format is
```yaml
deb [arch=i386 trusted=yes another=whatever] http://deb.debian.org/debian stable contrib non-free main
```
and not
```yaml
deb [arch=i386] [trusted=yes another=whatever] http://deb.debian.org/debian stable contrib non-free main
```
so I think that we should merge `r_arch` and `r_opts` if the latter is set. Perhaps we can turn it a hash parameter, like
```yaml
...
opts:
  trusted: 'yes'
  another: whatever
...
```
```python
{% set r_opts = '' %}

{%- set r_arch = 'arch=' ~ args.arch|join(',') if args.arch is defined else '' %}
{% if args.opts is defined %}
  {% if args.opts is string %}
    {% set r_opts = args.opts %}
  {% else %}
    {%- for k, v in args.opts.items() %}
      {% set r_opts = r_opts ~ ' ' ~ k ~ '=' ~ v %}
    {%- endfor %}
  {% endif %}
{% endif %}

{% if r_arch != '' or r_opts ! = '' %}
  {% set r_options = '[' ~ r_arch ~ ' ' ~ r_opts ~ ']'
{% else %}
  {% set r_options = ''
{% endif %}
...
...
  pkgrepo.managed:
    - name: {{ r_type }} {{ r_options }} {{ r_url }} {{ r_distro }} {{ r_comps }}
```

what do you think? Haven't tested this, is just a 'before my 1st. coffee' idea, so maybe it's wrong 😄

Comment (arthurlogilab) [reply apt/repositories.sls]: @javierbertoli looks good, I hadn't seen that bit of the documentation but you're absolutely right. I'm going to try out your code.

Comment (arthurlogilab) [reply apt/repositories.sls]: Appart from small typos, it works for the string pillar, but not the dict pillar (the  `for k, v in args.opts.items()` doesn't seem to work. Is it a jinja local variable problem ?

File: apt/repositories.sls [hunk 2]
```diff
@@ -26,8 +26,30 @@ debian-archive-keyring:
...
+
+{% set r_opts = '' %}
+{%- set r_arch = 'arch=' ~ args.arch|join(',') if args.arch is defined else '' %}
+{% if args.opts is defined %}
+  {% if args.opts is string %}
+    {% set r_opts = args.opts %}
+  {% else %}
+    {% set r_opts_list = [] %}
+    {%- for k, v in args.opts.items() %}
+      {% do r_opts_list.append(' ' ~ k ~ '=' ~ v) %}
```
Comment (javierbertoli) [inline apt/repositories.sls]: As you're converting the hash into a list, you can skip the initial whitespace here, so it'd be
```python
 {% do r_opts_list.append(k ~ '=' ~ v) %}
```

Comment (arthurlogilab) [reply apt/repositories.sls]: fixed, thanks

**Comment by javierbertoli** [comment]
Thanks, @arthurlogilab!! Great work!

**Pull request merged**